test/lib/ansible_test/_internal/commands/sanity/validate_modules.py

```diff
@@ -109,37 +134,48 @@ def test(self, args, targets, python):  # type: (SanityConfig, SanityTargets, Py
             else:
                 display.warning('Cannot perform module comparison against the base branch because the base branch was not detected.')
 
-        try:
-            stdout, stderr = run_command(args, cmd, env=env, capture=True)
-            status = 0
-        except SubprocessError as ex:
-            stdout = ex.stdout
-            stderr = ex.stderr
-            status = ex.status
+        errors = []
+        for plugin_type, plugin_targets in sorted(target_per_type.items()):
+            paths = [target.path for target in plugin_targets]
+            plugin_cmd = list(cmd)
+            if plugin_type != 'module':
+                plugin_cmd += ['--plugin-type', plugin_type]
+            plugin_cmd += paths
 
-        if stderr or status not in (0, 3):
-            raise SubprocessError(cmd=cmd, status=status, stderr=stderr, stdout=stdout)
+            try:
+                stdout, stderr = run_command(args, plugin_cmd, env=env, capture=True)
+                status = 0
+            except SubprocessError as ex:
+                stdout = ex.stdout
+                stderr = ex.stderr
+                status = ex.status
 
-        if args.explain:
-            return SanitySuccess(self.name)
+            if stderr or status not in (0, 3):
+                raise SubprocessError(cmd=plugin_cmd, status=status, stderr=stderr, stdout=stdout)
 
-        messages = json.loads(stdout)
+            if args.explain:
+                continue
 
-        errors = []
+            messages = json.loads(stdout)
+
+            plugin_errors = []
+            for filename in messages:
+                output = messages[filename]
 
-        for filename in messages:
-            output = messages[filename]
+                for item in output['errors']:
+                    plugin_errors.append(SanityMessage(
+                        path=filename,
+                        line=int(item['line']) if 'line' in item else 0,
+                        column=int(item['column']) if 'column' in item else 0,
+                        code='%s' % item['code'],
+                        message=item['msg'],
+                    ))
 
-            for item in output['errors']:
-                errors.append(SanityMessage(
-                    path=filename,
-                    line=int(item['line']) if 'line' in item else 0,
-                    column=int(item['column']) if 'column' in item else 0,
-                    code='%s' % item['code'],
-                    message=item['msg'],
-                ))
+            plugin_errors = settings.process_errors(plugin_errors, paths)
+            errors += plugin_errors
```

Comment:
```suggestion
            errors += settings.process_errors(plugin_errors, paths)
```

Reply:
Fixed in 3673f6055ce08b11653dae65a76a35b84fe0f555.